subsys/pm/device.c

```diff
@@ -167,6 +169,11 @@ bool pm_device_is_any_busy(void)
 	return false;
 }
 
+bool pm_device_action_can_block(void)
+{
+	return pm_state_next_get(_current_cpu->id)->state < PM_STATE_SUSPEND_TO_IDLE;
```

Comment:
when system is resumed, the next state will always be reported as active, a device will never be called outside of system PM with the next state being `PM_STATE_RUNTIME_IDLE` because system will have resumed first (I think, system PM code is a bit messy and not easy to follow). 

some comments about this logic would help on understanding what's going on here

Reply:
The logic was intended to be that if the CPU is in ACTIVE or RUNTIME_IDLE mode it would be ok to block.
If this is not the case, then I need someone to explain exactly when these transitions can't block, because otherwise I'm just poking in the dark @ceolin

Reply:
> I am fine in having this function inside the pm subsystem, but this looks to me as a qutie generic kernel function.

I agree, this is a general query regarding the kernel state, so it belongs there